pytorch_lightning/trainer/properties.py

```diff
@@ -187,6 +188,20 @@ def weights_save_path(self) -> str:
             return os.path.normpath(self._weights_save_path)
         return self._weights_save_path
 
+    @property
+    def checkpoint_callback(self) -> Optional[ModelCheckpoint]:
+        """
+        The first checkpoint callback in the Trainer.callbacks list, or ``None`` if
+        no checkpoint callbacks exist.
+        """
+        callbacks = self.checkpoint_callbacks
+        return callbacks[0] if len(callbacks) > 0 else None
```

Comment:
Is this required or just for convenience?

Thinking as a noob user. If I saw `trainer.checkpoint_callback` I would think it's the only one available, when in reality it's the first one.

If possible, I suggest renaming it to `first_checkpoint_callback` or, even better, removing it. If we want to support multiple callbacks, why have a property for the first one when order is not supposed to matter?

Reply:
yes, good observation. 
I made this mainly for compatibility with a few places where Trainer needs to access the checkpoint callback. One example is the test method which can load the best model, and this is done through the  best model path saved in the checkpoint callback. Right now, this is for convenience and backward compatibility. 

I propose to keep this at least until deprecation of of ModelCheckpoint through checkpoint_callback argument expires.
In the meantime, we can find a way to handle multiple ckpt callbacks all around the trainer in a consistent way.